addons/product/models/product_pricelist_item.py

```diff
@@ -177,22 +177,46 @@ def _compute_name_and_price(self):
                     item.env, item.fixed_price, monetary=True, dp="Product Price", currency_obj=item.currency_id)
             elif item.compute_price == 'percentage':
                 percentage = self._get_integer(item.percent_price)
-                item.price = _("%s %% discount", percentage)
-            else:
-                discount_type, percentage = self._get_displayed_discount(item)
-                if not (surcharge := item.price_surcharge):
+                if item.base_pricelist_id:
                     item.price = _(
-                        "%(percentage)s %% %(discount_type)s",
+                        "%(percentage)s %% discount on %(pricelist)s",
                         percentage=percentage,
-                        discount_type=discount_type,
+                        pricelist=item.base_pricelist_id.display_name
                     )
                 else:
                     item.price = _(
-                        "%(percentage)s %% %(discount_type)s and %(price)s surcharge",
-                        percentage=percentage,
-                        price=surcharge,
-                        discount_type=discount_type,
+                        "%(percentage)s %% discount on sales price",
+                        percentage=percentage
                     )
+            else:
+                base_str = ""
+                if item.base == 'list_price':
+                    base_str = "on sales price"
+                elif item.base == 'standard_price':
+                    base_str = "on product cost"
+                elif item.base == 'pricelist' and item.base_pricelist_id:
```

Comment:
Don't we need any string for base == pricelist and not base_pricelist_id? :thinking:

Reply:
![image](https://github.com/user-attachments/assets/acf3a333-9ad8-40b9-b77e-65913bbc0d4e)
No, there won't be such a case where `base == 'pricelist' and not base_pricelist_id`

Reply:
And then why it is elif not else :thinking: (as there are only 3 selection values)

Reply:
I thought validating using `elif` would be good, but we can make it work with `else` too. Should I use `else` instead of `elif`?

Reply:
Best thing would be add this in if then standard_price in elif and then else so if there is any case like above then in else base_str would be "on sales price"  like it is for percentage case